compat.cmake

```diff
@@ -0,0 +1,14 @@
+
+# old gcc-4.8 support, injection is required to compile gtest
+if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND
+  (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 4.8) AND
+  (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0))
+
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -include ${CMAKE_CURRENT_SOURCE_DIR}/src/c4/compat/gcc-4.8.hpp")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -include ${CMAKE_CURRENT_SOURCE_DIR}/src/c4/compat/gcc-4.8.hpp")
+
+  # c++17 compiler required
+  set(C4CORE_BUILD_BENCHMARKS OFF CACHE BOOL "" FORCE)
+  # LLVM required
+  set(C4CORE_SANITIZE OFF CACHE BOOL "" FORCE)
+endif()
```

Comment:
My initial reaction was to prefer your initial approach. But on reflection, I think this newer approach makes sense. This means it will be possible to compile any client code, nice. Just maybe remove the mention to gtest in the comment at the top, since it's no longer about that.

Reply:
Made a change let me know if you want something different.